src/samplers/activelearner.py

```diff
@@ -0,0 +1,318 @@
+    """
+    Active Learner using BMDAL library for Active Learning.
+
+    Args:
+        total_budget (int): Total budget for the active learning process.
+        parser_kwargs (dict): Keyword arguments for the parser.
+        model_kwargs (dict): Keyword arguments for the model.
+        train_kwargs (dict): Keyword arguments for training.
+
+    Attributes:
+        total_budget (int): Total budget for the active learning process.
+        bounds (list): Bounds for the parameter space.
+        parameters (list): Names of the parameters.
+        aquisition_batch_size (int): Batch size for acquisition.
+        init_num_samples (int): Number of initial samples.
+        selection_method (str): Method for selecting samples.
+        kernel_transform (list): Kernel transformation methods.
+        base_kernel (str): Base kernel for BMDAL.
+        parser_kwargs (dict): Keyword arguments for the parser.
+        model_kwargs (dict): Keyword arguments for the model.
+        train_kwargs (dict): Keyword arguments for training.
+        metrics (list): List to store metrics.
+    """
+
+    sampler_interface = S.ACTIVE
+
+    def __init__(
+        self,
+        total_budget: int,
+        parser_kwargs,
+        model_kwargs,
+        train_kwargs,
+        *args,
+        **kwargs,
+    ):
+
+        # fmt: off
+        self.total_budget          = total_budget
+        self.bounds                = kwargs.get('bounds', [None])
+        self.parameters            = kwargs.get('parameters', ['NA']*len(self.bounds))
+        self.aquisition_batch_size = kwargs.get('aquisition_batch_size', 512)
+        self.init_num_samples      = kwargs.get('num_initial_points', self.aquisition_batch_size)
+        self.selection_method      = kwargs.get('selection_method', 'random')
+        self.kernel_transform      = kwargs.get('kernel_transform', [])
+        self.base_kernel           = kwargs.get('base_kernel', 'll')
+
+        self.parser_kwargs         = parser_kwargs
+        self.model_kwargs          = model_kwargs
+        self.train_kwargs          = train_kwargs
+        self.metrics = []
+
+        self.parser                = getattr(parsers, parser_kwargs.pop('type'))(**parser_kwargs)
+        # fmt: on
+
+    def get_initial_parameters(self):
+        """
+        Generates initial parameter samples.
+
+        Returns:
+            list: List of dictionaries containing initial parameter samples.
+
+        """
+        # TODO: more than just random samples
+        assert self.parameters[0] != "NA"
+        batch_samples = []
+        for _ in range(self.init_num_samples):
+            params = [
+                torch.distributions.Uniform(lb, ub).sample().item()
+                for (lb, ub) in self.bounds
+            ]
+            param_dict = dict(zip(self.parameters, params))
+            batch_samples.append(param_dict)
+        return batch_samples
+
+    def _get_new_idxs_from_pool(self, model, train, pool) -> torch.Tensor:
```

Comment:
I think it might be important to specify in the docstring that the pool and train data are expected to be normalised.